### ISSUE TITLE
[BUG] For Certain Activity Course: App Crashes After Clicking Images From The Uploads Page 

### ISSUE THREAD

**Issue by kiranofans** [WikiEducationFoundation/apps-android-wikiedudashboard#220]
**Summary:** 
App crashes after clicking images from the Uploads page of the Gender gap on Dutch-language Wikipedia activity course

**Steps to reproduce:** 
1. Go to Explore -> Active Courses -> Click Gender gap on Dutch-language Wikipedia ->Uploads
2. In the Uploads page, click any image -> App crashes

**System logs:**
```
2019-10-28 14:47:12.441 22081-22081/org.wikiedufoundation.wikiedudashboard.release E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.wikiedufoundation.wikiedudashboard.release, PID: 22081
    java.lang.RuntimeException: Failure from system
        at android.app.Instrumentation.execStartActivity(Instrumentation.java:1533)
        at android.app.Activity.startActivityForResult(Activity.java:4411)
        at androidx.fragment.app.FragmentActivity.startActivityForResult(FragmentActivity.java:676)
        at androidx.core.app.ActivityCompat.startActivityForResult(ActivityCompat.java:234)
        at androidx.fragment.app.FragmentActivity.startActivityFromFragment(FragmentActivity.java:791)
        at androidx.fragment.app.FragmentActivity$HostCallbacks.onStartActivityFromFragment(FragmentActivity.java:933)
        at androidx.fragment.app.Fragment.startActivity(Fragment.java:1185)
        at androidx.fragment.app.Fragment.startActivity(Fragment.java:1173)
        at org.wikiedufoundation.wikiedudashboard.ui.coursedetail.uploads.view.CourseUploadsFragment.openCourseDetail(CourseUploadsFragment.kt:124)
        at org.wikiedufoundation.wikiedudashboard.ui.coursedetail.uploads.view.CourseUploadsFragment.access$openCourseDetail(CourseUploadsFragment.kt:31)
        at org.wikiedufoundation.wikiedudashboard.ui.coursedetail.uploads.view.CourseUploadsFragment$onCreateView$1.invoke(CourseUploadsFragment.kt:70)
        at org.wikiedufoundation.wikiedudashboard.ui.coursedetail.uploads.view.CourseUploadsFragment$onCreateView$1.invoke(CourseUploadsFragment.kt:31)
        at org.wikiedufoundation.wikiedudashboard.ui.adapters.CourseUploadsRecyclerAdapter.onCourseUpdatedClicked(CourseUploadsRecyclerAdapter.kt:23)
        at org.wikiedufoundation.wikiedudashboard.databinding.ItemRvCourseUploadBindingImpl._internalCallbackOnClick(ItemRvCourseUploadBindingImpl.java:161)
        at org.wikiedufoundation.wikiedudashboard.generated.callback.OnClickListener.onClick(OnClickListener.java:11)
        at android.view.View.performClick(View.java:6308)
        at android.view.View$PerformClick.run(View.java:23969)
        at android.os.Handler.handleCallback(Handler.java:751)
        at android.os.Handler.dispatchMessage(Handler.java:95)
        at android.os.Looper.loop(Looper.java:154)
        at android.app.ActivityThread.main(ActivityThread.java:6823)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1563)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1451)
     Caused by: android.os.TransactionTooLargeException: data parcel size 2773276 bytes
        at android.os.BinderProxy.transactNative(Native Method)
        at android.os.BinderProxy.transact(Binder.java:628)
        at android.app.ActivityManagerProxy.startActivity(ActivityManagerNative.java:3586)
        at android.app.Instrumentation.execStartActivity(Instrumentation.java:1526)
```

**Device and Android version:** 
Samsung Galaxy Tab A, Model: SM-T350
Android 7.1 Nougat

**Would you like to work on the issue?**
No

**Comment by kulloveth** [comment]
@kiranofans this issue occurs periodically which I think is when the app is kinda freezing, when you clear from background and restart is works fine again. will look into it anyways.

**Comment by kiranofans** [comment]
> @kulloveth this issue occurs periodically which I think is when the app is kinda freezing, when you clear from background and restart is works fine again. will look into it anyways.
So what if we clear from the background and restart programmatically. I guess this is related to the fragment lifecycle.

**Comment by kulloveth** [comment]
@kiranofans  so according to the error log, its either  the image is too large for the intent that is passing it or its also passing another item with the image which is too large for it. still looking into it

**Comment by kulloveth** [comment]
@ujjwalagrawal17  as regards to the logs showing during this crash, my interpretation of the error is that the files been passed is too large. Serializable takes time to convert Objects and in the process creates more temp objects. I suggest the model is converted using parcellable which I think will end the crash. The only disadvantage I am thinking of now is its consequences as we are working on persisting the data. Waiting for your opinion so I can go ahead with the conversion or a better suggestion.

**Comment by ujjwalagrawal17** [comment]
Yup, I also think that should be the reason. We can try changing the serializable to parcellable to see if it fixes the issue.

**Comment by kulloveth** [comment]
> Yup, I also think that should be the reason. We can try changing the serializable to parcellable to see if it fixes the issue.

done with the conversion,observing it out for a while before I PR

**Comment by kulloveth** [comment]
still looking for a fix for this after the parcel size increased from 2mb to 5mb when I changed to Parcellable and according to this https://developer.android.com/reference/android/os/TransactionTooLargeException you cant transfer more than 1mb through a bundle
it turns out parcellable is not the best afterall like myself and most people thought, trying out this experiment https://medium.com/nemanja-kovacevic/yet-another-post-on-serializable-vs-parcelable-1e829e98472 proved that too

**Comment by kiranofans** [comment]
> looking

I'm not sure about this, but maybe we can find a way to compress or restrict the images' size using the glide or similar library?

**Comment by kulloveth** [comment]
> > looking
> 
> I'm not sure about this, but maybe we can find a way to compress or restrict the images' size using the glide or similar library?

glide is already in  use and its not about the image since I have commented the code for image and even though its not showing it still crashes, this happens specifically for some items on the list. I have pointed out Item 1 and 3. even when the uploads are loading it takes so much time and in the process leakcanary will come up dumping heap. its something about lists and arrays.
